src/Materials/material.ts

```diff
@@ -746,6 +787,21 @@ export class Material implements IAnimatable {
         if (this._scene.useMaterialMeshMap) {
             this.meshMap = {};
         }
+
+        Material.OnEventObservable.notifyObservers(this, MaterialEvent.Created);
+
+        // TODO: _defineNamesFromPlugins needs to be updated with `MaterialPluginManager.AddPluginToMaterial` too

```

Comment:
same

Reply:
holding on the decision of whether registration will be static or not to finish this one.